bundle/internal/schema/main.go

```diff
@@ -112,37 +113,56 @@ func makeVolumeTypeOptional(typ reflect.Type, s jsonschema.Schema) jsonschema.Sc
 }
 
 func main() {
-	if len(os.Args) != 2 {
-		fmt.Println("Usage: go run main.go <output-file>")
+	if len(os.Args) != 3 {
+		fmt.Println("Usage: go run main.go <work-dir> <output-file>")
 		os.Exit(1)
 	}
 
+	// Directory with annotation files
+	workdir := os.Args[1]
 	// Output file, where the generated JSON schema will be written to.
-	outputFile := os.Args[1]
+	outputFile := os.Args[2]
+
+	generateSchema(workdir, outputFile)
+}
+
+func generateSchema(workdir, outputFile string) {
+	annotationsPath := path.Join(workdir, "annotations.yml")
+	annotationsOpenApiPath := path.Join(workdir, "annotations_openapi.yml")
+	annotationsOpenApiOverridesPath := path.Join(workdir, "annotations_openapi_overrides.yml")
```

Comment:
I think we should limit the scope of custom annotations to only non-OpenAPI overrides. I don't see much if any overrides for OpenAPI fields. Any such one-of overrides can be easily be added similar to `removeJobsFields`.

Reply:
My reasoning is that `annotations_openapi.yml` and `annotations_openapi_overrides.yml` are large files that serve little purpose and complicate understanding how OpenAPI descriptions are loaded onto the schema.

Reply:
We should be cautious about overexposing here since this will be harder to maintain as we launch SDK v1 with versioning per package. The interface of `PkgName + TypeName` might not be the best one for structs outside the CLI repo.

> It seems valid to override in this case since it affects json-schema usability

Yeah, but we should not maintain docs for multiple API fields that we do not own. One-off overrides can be achieved via `removeJobsFields`.

Reply:
I'd like to keep it around but as minimal as possible. Good to have an escape hatch.

Reply:
> Could you elaborate more about SDK v1, how it will change json-schema, will it also include different versions?

IIRC we'll have different Go packages for each version, `jobsv1`, `jobsv2` etc so the package name in the key might no longer correspond to the new package names.

Reply:
We embed these types and we infer their path from the actual struct.

The only issue here is that we add another place where we depend on these paths and that we need to update when this move happens. That's a reasonable cost and worth it, IMO.